server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java

```diff
@@ -3470,6 +3480,42 @@ protected ServiceOfferingVO createServiceOffering(final long userId, final boole
         }
     }
 
+    /**
+     * This method will return valid and non-empty expiryAction  when
+     * "instance.lease.enabled" feature is enabled at global level
+     * leaseDuration is positive > -1 and has valid leaseExpiryAction provided or configured
+     * @param leaseDuration
+     * @param cmdExpiryAction
+     * @return leaseExpiryAction
+     */
+    public static String validateAndGetLeaseExpiryAction(Long leaseDuration, String cmdExpiryAction) {
+        String leaseExpiryAction = null;
+        if (!VMLeaseManagerImpl.InstanceLeaseEnabled.value()
+                || (leaseDuration == null && StringUtils.isEmpty(cmdExpiryAction))) { // both are null
```

Comment:
Could cmdExpiryAction be passed as an empty string (not null) or is invoked with null values also? This method would be catching both cases. If only checking for null, can we use `ObjectUtils.allNull(leaseDuration, cmdExpiryAction)` instead?